### PR TITLE
Spheron Terraform Provider

### PR DESCRIPTION
**Pull request by ilij4**

### CODE REVIEW

File: pages/terraform.mdx [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+|Ignis Large     |8  |32Gi  |
+|Ventus Large    |2  |8Gi   |
+|Terra Medium    |4  |8Gi   |
+|Ventus 2x Medium|2  |4Gi   |
+|Terra 3x Large  |10 |32Gi  |
+|Ventus 2x Large |4  |8Gi   |
+|Terra Small 1Ti |4  |4Gi   |
+|Terra Large Ti  |8  |16Gi  |
+
+#### Available regions
```
Comment (rekpero) [inline pages/terraform.mdx:222]: same with this, we have same content at the end

File: pages/terraform.mdx [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+- `memory` (String) Instance Memory in GB.
+- `persistent_storage` (Attributes) Persistent storage that will be attached to the instance. (see [below for nested schema](#nestedatt--persistent_storage))
+
+Read-Only:
+
+- `id` (String) Id or the instance.
+- `ports` (Attributes List) The list of port mappings (see [below for nested schema](#nestedatt--ports))
+
+<a id="nestedatt--env"></a>
+### Nested Schema for `env`
```
Comment (rekpero) [inline pages/terraform.mdx:280]:
```suggestion
#### Nested Schema for `env`
```

File: pages/terraform.mdx [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+
+Required:
+
+- `class` (String) Storage class. Available classes are HDD, SSD and NVMe
+- `mount_point` (String) Attachement point used fot attaching persistent storage.
+- `size` (Number) Persistent storage in GB. Value cannot exceed 1024GB
+
+
+<a id="nestedatt--ports"></a>
+### Nested Schema for `ports`
```
Comment (rekpero) [inline pages/terraform.mdx:299]:
```suggestion
#### Nested Schema for `ports`
```

File: pages/terraform.mdx [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+### Nested Schema for `env`
+
+Required:
+
+- `key` (String) Environment variable key.
+- `value` (String) Environment variable value.
+
+
+<a id="nestedatt--persistent_storage"></a>
+### Nested Schema for `persistent_storage`
```
Comment (rekpero) [inline pages/terraform.mdx:289]:
```suggestion
#### Nested Schema for `persistent_storage`
```

File: pages/terraform.mdx [hunk 5]
```diff
@@ -0,0 +1,425 @@
...
+|MassBit Route Gateway|TOKEN, GW_ID         |
+|Arbitrum One Node|                     |
+|Filecoin Mainnet Node|                     |
+|Shardeum Testnet Validator|EXT_IP, INT_IP, EXISTING_ARCHIVERS, APP_MONITOR, DASHPASS, DASHPORT, SERVERIP, LOCALLANIP, SHMEXT, SHMINT|
+
+
+#### Available machine images
```
Comment (rekpero) [inline pages/terraform.mdx:333]:
```suggestion
### Available machine images
```

File: pages/terraform.mdx [hunk 6]
```diff
@@ -0,0 +1,425 @@
...
+|Ventus Large    |2  |8Gi   |
+|Terra Medium    |4  |8Gi   |
+|Ventus 2x Medium|2  |4Gi   |
+|Terra 3x Large  |10 |32Gi  |
+|Ventus 2x Large |4  |8Gi   |
+|Terra Small 1Ti |4  |4Gi   |
+|Terra Large Ti  |8  |16Gi  |
+
+
+#### Available regions
```
Comment (rekpero) [inline pages/terraform.mdx:360]:
```suggestion
### Available regions
```

File: pages/terraform.mdx [hunk 7]
```diff
@@ -0,0 +1,425 @@
...
+
+<a id="nestedatt--ports"></a>
+### Nested Schema for `ports`
+
+Read-Only:
+
+- `container_port` (Number) Container port that will be exposed.
+- `exposed_port` (Number) The port container port will be exposed to. Exposed port will be know and available for use after the deployment.
+
+#### Awailable marketplace apps:
```
Comment (rekpero) [inline pages/terraform.mdx:306]:
```suggestion
#### Available marketplace apps
```

File: pages/terraform.mdx [hunk 8]
```diff
@@ -0,0 +1,425 @@
...
+- `env` (Attributes Set) The list of environmetnt variables. (see [below for nested schema](#nestedatt--env))
+- `env_secret` (Attributes Set) The list of secret environmetnt variables. (see [below for nested schema](#nestedatt--env_secret))
+- `health_check` (Attributes) Path and container port on which health check should be done. (see [below for nested schema](#nestedatt--health_check))
+- `id` (String) Id of the instance.
+- `machine_image` (String) Machine image name which should be used for deploying instance.
+- `memory` (String) Instance Memory in GB.
+- `persistent_storage` (Attributes) Persistent storage that will be attached to the instance. (see [below for nested schema](#nestedatt--persistent_storage))
+
+<a id="nestedatt--env"></a>
+Nested Schema for `env`
```
Comment (rekpero) [inline pages/terraform.mdx:152]:
```suggestion
#### Nested Schema for `env`
```

File: pages/terraform.mdx [hunk 9]
```diff
@@ -0,0 +1,425 @@
...
+
+Required:
+
+- `key` (String) Environment variable key.
+- `value` (String) Environment variable value.
+
+
+<a id="nestedatt--env_secret"></a>
+Nested Schema for `env_secret`
```
Comment (rekpero) [inline pages/terraform.mdx:161]:
```suggestion
#### Nested Schema for `env_secret`
```

File: pages/terraform.mdx [hunk 10]
```diff
@@ -0,0 +1,425 @@
...
+
+Required:
+
+- `key` (String) Environment variable key.
+- `value` (String) Environment variable value.
+
+<a id="nestedatt--ports"></a>
+Nested Schema for `ports`
```
Comment (rekpero) [inline pages/terraform.mdx:169]:
```suggestion
#### Nested Schema for `ports`
```

File: pages/terraform.mdx [hunk 11]
```diff
@@ -0,0 +1,425 @@
...
+Required:
+
+- `container_port` (Number) Container port that will be exposed.
+
+Optional:
+
+- `exposed_port` (Number) The port container port will be exposed to. Currently only posible to expose to port 80. Leave empty to map to random value. Exposed port will be know and available for use after the deployment.
+
+<a id="nestedatt--health_check"></a>
+Nested Schema for `health_check`
```
Comment (rekpero) [inline pages/terraform.mdx:180]:
```suggestion
#### Nested Schema for `health_check`
```

**Review by rekpero — CHANGES_REQUESTED** [review]
check my RC @ilij4

File: pages/terraform.mdx [hunk 12]
```diff
@@ -0,0 +1,425 @@
...
+Guide on how to use the Spheron Terraform provider.
+
+## Prerequisites
+
+Before you begin, please ensure that you have the following prerequisites installed on your machine:
+
+- Terraform (version 0.13 or later)
+- Spheron provider plugin (version 1.0.0 or later)
+
+## Installation
```
[the remark this answers is not in the document]
Comment (ilij4) [reply pages/terraform.mdx:15]: @ it is at the bottom

[the remark this answers is not in the document]
Comment (rekpero) [reply pages/terraform.mdx:15]: just put the Terraform link in the installation as well

**Review by rekpero — CHANGES_REQUESTED** [review]
some small update needed

**Review by aayushmahapatra — APPROVED** [review]
Lgtm

**Pull request merged**